OSDOCS-11555:Configuring network observability network policy (#79960)
openshift/openshift-docs: modules/network-observability-deploy-network-policy.adoc

// Module included in the following assemblies:

// * networking/network_observability/network-observability-network-policy.adoc

:_mod-docs-content-type: PROCEDURE
[id="network-observability-deploy-network-policy_{context}"]
= Configuring an ingress network policy by using the FlowCollector custom resource

> You can configure the `FlowCollector` custom resource to deploy an ingress Network Policy for Network Observability by setting the `spec.NetworkPolicy.enable` specification to `true`. By default, the specification is `false`.

Reviewer: It seems, after a quick Atom search, that _network policy_ is lower-case (???).
```suggestion
You can configure the `FlowCollector` custom resource to deploy an ingress network policy for Network Observability by setting the `spec.NetworkPolicy.enable` specification to `true`. By default, the specification is `false`.
```

> Ensure that Network Observability components you have configured can communicate with components that are external to the namespace, which might include:

Reviewer: ```suggestion
Ensure that the Network Observability components that you have configured can communicate with components that are external to the namespace, which might include:
```

> Ensure that Network Observability components you have configured can communicate with components that are external to the namespace, which might include:

Reviewer 2: maybe reformulate to make it more obvious this is only for a special case? Something like:

```suggestion
If you have installed Loki, Kafka or any exporter in a different namespace that also has a network policy, you must ensure that the Network Observability components can communicate with them.
```
?

Author: OK I like this suggestion actually its more clear than just linking.

>    * Connection to Loki (as defined in `FlowCollector` `spec.loki`)
>    * Connection to Kafka (as defined in `FlowCollector` `spec.kafka`)
>    * Connection to any exporter (as defined in FlowCollector `spec.exporters`)

Reviewer: Is this right? It seems like we need more here. 
```suggestion
   * Connection to Loki (as defined in the `FlowCollector` CR `spec.loki` parameter)
   * Connection to Kafka (as defined in the `FlowCollector` CR `spec.kafka` parameter)
   * Connection to any exporter (as defined the `FlowCollector` CR `spec.exporters` parameter)
```
Will people know what this means or should we provide examples?

Author: I think NetObserv users should be pretty familiar with the different storage/connection options, i.e. Loki and Kafka. We have topics about Exporters and also configuring Loki and Kafka are part of the initial installation of the Operator. Maybe its worth a link over there in case there is a question.

   * If you are using Loki and including it in the policy target, connection to an external object storage (as defined in your `LokiStack` related secret)

.Procedure

> . . In the web console, navigate to *Operators* -> *Installed Operators*.

Reviewer: Per ISG: [Use navigate only when a more direct verb, such as click, select, browse, or go to, is not applicable](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#n)
```suggestion
. . In the web console, go to the *Operators* -> *Installed Operators* page..
```

. Under the *Provided APIs* heading for *Network Observability*, select *Flow Collector*.
. Select *cluster* then select the *YAML* tab.

> . Configure the `FlowCollector` custom resource. A sample configuration is as follows:

Reviewer: Minimalism
```suggestion
. Configure the `FlowCollector` CR. A sample configuration is as follows:
```

+
[id="network-observability-flowcollector-configuring-network-policy_{context}"]

> .Configure `FlowCollector` for network policy

Reviewer: ```suggestion
.Example`FlowCollector` CR for network policy
```

// Module included in the following assemblies:

// * networking/network_observability/network-observability-network-policy.adoc

:_mod-docs-content-type: PROCEDURE
[id="network-observability-deploy-network-policy_{context}"]
= Configuring an ingress network policy by using the FlowCollector custom resource

> You can configure the `FlowCollector` custom resource to deploy an ingress Network Policy for Network Observability by setting the `spec.NetworkPolicy.enable` specification in  to `true`. By default, the specification is `false`.

Reviewer 3: Can remove the `in` after specification and the extra space

Ensure that Network Observability components you have configured can communicate with components that are external to the namespace, which might include:

   * Connection to Loki (as defined in `FlowCollector` `spec.loki`)
   * Connection to Kafka (as defined in `FlowCollector` `spec.kafka`)
   * Connection to any exporter (as defined in FlowCollector `spec.exporters`)
   * If you are using Loki and including it in the policy target, connection to an external object storage (as defined in your `LokiStack` related secret)

> If you need further isolation for egress traffic, you can create an additional policy that. For more information, see 

Reviewer 3: missing `for` after policy

Author: I actually need to remove this entire thing because we're not going to recommend that anymore, per Joel's comment here: https://issues.redhat.com/browse/OSDOCS-11555

// Module included in the following assemblies:

// * networking/network_observability/network-observability-network-policy.adoc

:_mod-docs-content-type: PROCEDURE
[id="network-observability-deploy-network-policy_{context}"]

> = Configuring a network policy by using the FlowCollector custom resource

Reviewer 3: Can include ingress in the heading `Configuring a ingress network policy by using ...`

> You can configure the `FlowCollector` custom resource to deploy a Network Policy for Network Observability by setting the `spec.deployNetworkPolicy.enable` specification in  to `true`. By default, the specification is `false`.

Reviewer 3: Same here; `to deploy a ingress Network Policy for Network Observability`

> You can configure the `FlowCollector` custom resource to deploy a Network Policy for Network Observability by setting the `spec.deployNetworkPolicy.enable` specification in  to `true`. By default, the specification is `false`.

Reviewer 2: also it's `spec.networkPolicy.enable` not `spec.deployNetworkPolicy.enable`

   * Connection to Loki (as defined in `FlowCollector` `spec.loki`)
   * Connection to Kafka (as defined in `FlowCollector` `spec.kafka`)
   * Connection to any exporter (as defined in FlowCollector `spec.exporters`)
   * If you are using Loki and including it in the policy target, connection to an external object storage (as defined in your `LokiStack` related secret)

If you need further isolation for egress traffic, you can create an additional policy that. For more information, see 
//"Creating an egress network policy for Network Observability".

.Procedure
. . In the web console, navigate to *Operators* -> *Installed Operators*.
. Under the *Provided APIs* heading for *Network Observability*, select *Flow Collector*.
. Select *cluster* then select the *YAML* tab.
. Configure the `FlowCollector` custom resource. A sample configuration is as follows:
+
[id="network-observability-flowcollector-configuring-network-policy_{context}"]
.Configure `FlowCollector` for network policy
[source, yaml]
----
apiVersion: flows.netobserv.io/v1beta2
kind: FlowCollector
metadata:
  name: cluster
spec:
  namespace: netobserv

>   deploymentModel: Direct

Reviewer 3: can remove the deploymentModel line. Customer might feel its only Direct mode and not for Kafka mode

  networkPolicy:
    enable: true   <1>

>     additionalNamespaces: [] <2>

Reviewer 2: I would leave the default values in the example, because if the user replaces them by an empty list, that would break a couple of things.

```suggestion
    additionalNamespaces: ["openshift-console", "openshift-monitoring"] <2>
```
